src/main/java/com/uber/cadence/internal/worker/ActivityWorker.java

```diff
@@ -152,55 +174,94 @@ private TaskHandlerImpl(ActivityTaskHandler handler) {
 
     @Override
     public void handle(
-        IWorkflowService service, String domain, String taskList, PollForActivityTaskResponse task)
+        IWorkflowService service, String domain, String taskList, MeasurableActivityTask task)
```

Comment:
I don't see why the interface change is needed. The stopwatch can be started and stopped in the scope of this method.

Reply:
This is to report e2e activity task latency. So the stopwatch is created at the beginning of the poll method and then stopped here.

Reply:
I still think this change is ugly. Would you measure it directly in the PollTask.run method?